msg/versioned/ArmingCheckReply.msg

```diff
@@ -1,34 +1,42 @@
+# Arming check reply.
+# 
+# This is a response to an ArmingCheckRequest message sent by the FMU to an external component, such as a ROS 2 navigation mode.
+# The response contains the current set of external mode requirements, and a queue of events indicating recent failures to set the mode (which the FMU may then forward to a ground station).
+# The request is sent regularly to all registered ROS modes, even while armed, so that the FMU always knows and can forward the current state.
+#
+# Note that the external component is identified by its registration_id, which is allocated to the component during registration (arming_check_id in RegisterExtComponentReply).
+# The message is not used by internal/FMU components, as their mode requirements are known at compile time.
+
 uint32 MESSAGE_VERSION = 0
 
-uint64 timestamp # time since system start (microseconds)
+uint64 timestamp # [us] Time since system start.
 
-uint8 request_id
-uint8 registration_id
+uint8 request_id # Id of ArmingCheckRequest for which this is a response.
+uint8 registration_id # Id of external component emitting this response.
 
-uint8 HEALTH_COMPONENT_INDEX_NONE = 0
+uint8 HEALTH_COMPONENT_INDEX_NONE = 0 # Index of health component for which this response applies.
 
-uint8 health_component_index      # HEALTH_COMPONENT_INDEX_*
-bool health_component_is_present
-bool health_component_warning
-bool health_component_error
+uint8 health_component_index # [@enum HEALTH_COMPONENT_INDEX]
+bool health_component_is_present # Unused. Intended for use with health events interface (health_component_t in events.json).
+bool health_component_warning # Unused. Intended for use with health events interface (health_component_t in events.json).
+bool health_component_error # Unused. Intended for use with health events interface (health_component_t in events.json).
 
-bool can_arm_and_run              # whether arming is possible, and if it's a navigation mode, if it can run
+bool can_arm_and_run # True for navigation mode components if the vehicle meets all the requirements to arm in the mode or switch to the mode if it is already armed. For non-mode components, such as remote IDs, it acts as a general arming requirement.
```

Comment:
@bkueng  I am still not convinced by the first sentence. 

Essentially it means `"if it can run" == "or switch to the mode if it is already armed"`. Is that what you meant by "it can run"?

You could have been referring to the case where something is armed but still interlocked. I'm thinking here of a throw launch where the vehicle is armed but in a particular state where it is not active until an acceleration trigger is detected.

Assuming the text is correct, maybe this might be better.

```suggestion
bool can_arm_and_run # True if the component can arm (otherwise false). For navigation mode components if the component can arm in the mode (or switch to the mode when armed).
```